nexus/pkg/artifacts/artifacts.go

```diff
@@ -46,6 +46,13 @@ type ManifestV1 struct {
 	Contents            map[string]ManifestEntry    `json:"contents"`
 }
 
+type ArtifactLinker struct {
```

Comment:
Nit: Always look for an opportunity to split code into multiple files. Linking is different than saving, so it should be separate.